### DISCUSSION TITLE
D1 Mini Tasmota Set Gpio D7 to output

### DISCUSSION THREAD

**Discussion by titou4307** [arendst/Tasmota#16954] [Q&A]
Hello
I am new bee on programmation Tasmota / Home Assistant....

I have a PCB micro relay on this for to pilot windows "shutter" on my home...

Three actions are possible down, up and stop
It only needeed to pass 
D5 to output for up
D6 to output for stop
D7 to output for down

How to pass this command to the D1 mini (under tasmota ??)

Many thanks

**Comment by sfromis** [comment]
Seen this? https://tasmota.github.io/docs/Blinds-and-Shutters/

**Comment by titou4307** [comment]
@sfromis 
Thanks for your fisrt response

I have maybe not give all the information of my request...
I have made research and it seems I need to pass a request (HTTP or MQTT) like into espeasy : 

http://192.168.xxx.xxx/control?cmd=GPIOtoggle,12

Does it possible to pass request to tasmota for to change the state of the GPIO12 (D5) 0 -> 1  (0 volt to 3,3 volts) ?

titou43 (from france)

**Reply by sfromis** [reply]
To have a pin available for simple control, config it as a Relay, and then you can use the Power command. All command are available via both MQTT (the primary method), console, http and serial. Still, Tasmota does support shutters instead of wanting to control the pins directly.

**Reply by titou4307** [reply]
Ok so I will be put GPIO12 as Relay

You could give me syntax for the power command ?

**Reply by sfromis** [reply]
https://tasmota.github.io/docs/Commands/#power
https://tasmota.github.io/docs/Commands/#with-web-requests

**Comment by titou4307** [comment]
Hello
I succeed to install all the hardware and the programmation
Many thanks for all your help !!

titou4307 (from France)

**Comment by titou4307** [comment]
Hello
I would like to upgrade my project on my PCB...

I want to po pilot a relay here : 

https://boutique.semageek.com/fr/1423-module-relais-5v-opto-isole-1-canaux-10a-3002643620642.html

My D1 Mini under tasmota, I have the GPIO here "busy" : 
GPIO12 : Relay
GPIO13 : Relay
GPIO14 : Relay

How can I connect the relay 1CH of this link ? I have try GPIO05 as Relay too but it is not OK

Best regard

**Reply by barbudor** [reply]
What is the exact schematic of this board?
If you don't have it, take a paper, a pencil, your multimeter and find out
You won't know how to properly connect without knowing how the board is made
